video/out/w32_common.c

```diff
@@ -1526,6 +1542,26 @@ static LRESULT CALLBACK WndProc(HWND hWnd, UINT message, WPARAM wParam,
             ((LPNCCALCSIZE_PARAMS) lParam)->rgrc[0].top -= get_title_bar_height(w32);
         }
         break;
+    case WM_IME_STARTCOMPOSITION: {
+        HIMC imc = ImmGetContext(w32->window);
+        COMPOSITIONFORM cf = {.dwStyle = CFS_POINT, .ptCurrentPos = {0, 0}};
+        ImmSetCompositionWindow(imc, &cf);
+        ImmReleaseContext(w32->window, imc);
+        break;
+    }
+    case WM_CREATE:
+        // The IME can only be changed to alphanumeric input after it's initialized.
+        // Unfortunately, there is no way to know when this happens, as
+        // none of the WM_CREATE, WM_INPUTLANGCHANGE, or WM_IME_* messages work.
+        // One can only pray that the IME is initialized within a short time after
+        // the window is created.
+        SetTimer(w32->window, (UINT_PTR)WM_CREATE, 250, NULL);
+        break;
+    case WM_TIMER:
+        if (wParam == WM_CREATE)
+            // Default to alphanumeric input when the IME is first initialized.
+            set_ime_conversion_mode(w32, IME_CMODE_ALPHANUMERIC);
+        return 0;
```

Comment:
Break if not handled.
```suggestion
        if (wParam == WM_CREATE) {
            // Default to alphanumeric input when the IME is first initialized.
            set_ime_conversion_mode(w32, IME_CMODE_ALPHANUMERIC);
            return 0;
        }
        break;
```

Regardless, do we have to set it with timer after WM_CREATE? Can't this be done, on first input to avoid the race?